final_project/Tests/ContactUsTests.cs

```diff
@@ -9,25 +11,28 @@ public class ContactUsTests : UITestFixture
         [Test]
         public async Task ContactUsTest()
         {
-            await page.GotoAsync("https://automationexercise.com/");
-            await page.GetByLabel("Consent", new() { Exact = true }).ClickAsync();
-            await page.GetByRole(AriaRole.Link, new() { Name = "Contact us" }).ClickAsync();
-            await page.GetByPlaceholder("Name").FillAsync("Mike");
-            await page.GetByPlaceholder("Email", new() { Exact = true }).FillAsync("bilikmike@gmail.com");
-            await page.GetByPlaceholder("Subject").FillAsync("test subject");
-            await page.Locator("input[name='upload_file']").SetInputFilesAsync(["../../../TestFiles/test_file.txt"]);
-            await page.GetByPlaceholder("Your Message Here").FillAsync("This is the test message for testing 'Contact Us' form");
-            page.Dialog += async (_, dialog) =>
-                {
-                    await dialog.AcceptAsync();
-                };
-            await page.WaitForLoadStateAsync(LoadState.DOMContentLoaded);
-            await page.GetByRole(AriaRole.Button, new() { Name = "Submit" }).ClickAsync();
-            await Assertions.Expect(page.Locator("#contact-page")).ToContainTextAsync("Success! Your details have been submitted successfully.");
-            await page.Locator(".btn-success").ClickAsync();
-            await Assertions.Expect(page.GetByRole(AriaRole.Heading, new() { Name = "AutomationExercise" })).ToBeVisibleAsync();
-        }
+            //Arrange
+            var homePage = new HomePage(page);
+            var contactUsPage = new ContactUsPage(page);
+            var contactInfo = new ContactInfo()
+            {
+                UserName = "John",
+                UserEmail = "testmail@gmail.com",
+                Subject = "test subject",
+                FilePath = "../../../TestFiles/test_file.txt",
+                Message = "This is the test message for testing 'Contact Us' form"
+            };
+
+            //Act
+            await homePage.OpenContuctUs();
+            await contactUsPage.FillContactInfo(contactInfo);
+            await contactUsPage.SubmitContactInfo();
 
+            //Assert
+            await Assertions.Expect(contactUsPage.GetContactInfoLocator()).ToContainTextAsync("Success! Your details have been submitted successfully.");
+            await contactUsPage.ClickSuccessButton();
+            StringAssert.EndsWith(homePage.homePageURL, page.Url);
```

Comment:
it is always better to check page content if possible (if Home page has some special element). Checking URL is almost useless as for me - you can't be sure that user sees something (and not a empty screen) and test will fail if URL changes (at the same time page content can be completely Ok)

Reply:
Done